WIP: Fix small issues on windows

bagage opened the pull request:
Fixes #261 

- I'm not sure about the clipboard stuff: maybe `clipboard` library is less powerful than using `xclip/pbsel` directly (binary data and such?). Thoughts on that @abl?
- Waiting for upstream on remaining issues for now.

richrd commented:
Since we already have some other dependencies like wcwidth I'm not against adding one for the clipboard. There's at least the libraries `clipboard` and `pyperclip`. Not sure which one we should use but `pyperclip` seems to be more popular.

bagage commented:
* I [investigated the init_pair(0) issue](https://github.com/zephyrproject-rtos/windows-curses/issues/10), for now I don't see any solution apart from not executing that code on Windows (until upstream provide a fix for that).
* For the ctrl+h issue, I see in suplemon that there are already some shortcuts with the same issue right now ([^I ^J ^M ^Z](https://github.com/richrd/suplemon/blob/master/suplemon/key_mappings.py#L84-L101)). Maybe help should be bind to F2 instead? Or maybe we can have some specific key mapping for windows only? You tell me!
* Clipboard handling is my next investigation :).

richrd commented:
- I think for now we can just not run `init_pair` on Windows if it doesn't work, as long as it doesn't cause other issues.
- Not quite sure what to do with ctrl+h. What happens currently when pressing backspace or ctrl+h? F2 can't be used because it's already bound to "reload file". If backspace works fine we could just remap help to some other key. Maybe we need a windows variant of the default keymap :thinking: 

Thanks again for the effort, I appreciate it! :)

bagage commented:
backspace is the same as hitting ctrl+h, so instead of removing the latest character, it toggles the help. Hence the shortcut for help toggle should be something else, maybe only for windows (F3?F7?F8?). But since default keymapping is global, I'm not sure how this could be done. 

Maybe it can be open as a separate issue as it does not block suplemon from working at least?

richrd commented:
Sorry for the late answer. Yeah, I guess it can be a separate issue, as long as backspace works for deleting characters. Help is also available by running the command `help` so it's not a huge issue. I do think we can have different default keymaps for different platforms as that's what Sublime seems to do.

richrd commented:
I took a closer look at `clipboard` and `pyperclip` and I think we should use `pyperclip` instead.
`clipboard` just wraps `pyperclp` and doesn't provide any additional functionality: https://github.com/terryyin/clipboard/blob/master/clipboard.py

bagage commented:
Hi @richrd, sorry for the long answer! I finally managed to find some time to address the remaining issues.

* replaced `clipboard` with `pyperclip`
* `^H` is now behaving as `backspace` instead of toggling help, even on Linux. Ideally if shortcuts could be per platform that would be needed only on Windows.
* replaced `resizeterm` with `resize_term` to make it work on Windows, while not breaking Linux (I tested quickly)

Let me know if there is anything to rework/address :).

bagage commented:
Bump @richrd :call_me_hand:

richrd commented:
Hi and sorry for the late response! Summer has been busy :)
I'll test this out next week. The code looks good so I think it's pretty much ready for merging. 
Thanks for your help :)
